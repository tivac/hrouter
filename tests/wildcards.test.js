"use strict";

const hrouter = require("../src/router.js");

describe("hrouter", () => {
    beforeEach(() => {
        hrouter._routes = Object.create(null);
    });

    describe("wildcards", () => {
        it("should register wildcard routes", () => {
            hrouter("/*", jest.fn());
            hrouter("*", jest.fn());
    
            expect(hrouter._routes).toMatchSnapshot();
        });

        it("should register nested wildcard routes", () => {
            hrouter("/fooga/*", jest.fn());

            expect(hrouter._routes).toMatchSnapshot();
        });
        
        it("should register wildcard routes w/ multiple callbacks", () => {
            hrouter("/*", jest.fn(), jest.fn(), jest.fn());
            hrouter("*", jest.fn(), jest.fn(), jest.fn());
    
            expect(hrouter._routes).toMatchSnapshot();
        });
        
        it("should register nested wildcard routes w/ multiple callbacks", () => {
            hrouter("/*", jest.fn(), jest.fn(), jest.fn());
            hrouter("/fooga/*", jest.fn(), jest.fn());
    
            expect(hrouter._routes).toMatchSnapshot();
        });
    
        it("should match simple wildcard routes", () => {
            const fn = jest.fn((ctx, next) => next());
    
            hrouter("/*", fn);
            hrouter("/fooga", fn);
    
            hrouter.go("/fooga");
    
            expect(fn.mock.calls).toMatchSnapshot();
        });
    
        it("should match nested wildcard routes", () => {
            const fn = jest.fn((ctx, next) => next());
    
            hrouter("/*", fn);
            hrouter("/fooga", fn);
            hrouter("/fooga/*", fn);
            hrouter("/fooga/booga", fn);
    
            hrouter.go("/fooga/booga");
    
            expect(fn.mock.calls).toMatchSnapshot();
        });
    });
});
