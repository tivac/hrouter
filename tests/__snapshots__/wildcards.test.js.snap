// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hrouter wildcards should match nested wildcard routes 1`] = `
Array [
  Array [
    Object {
      "url": "/fooga/booga",
    },
    [Function],
  ],
  Array [
    Object {
      "url": "/fooga/booga",
    },
    [Function],
  ],
  Array [
    Object {
      "url": "/fooga/booga",
    },
    [Function],
  ],
  Array [
    Object {
      "url": "/fooga/booga",
    },
    [Function],
  ],
]
`;

exports[`hrouter wildcards should match simple wildcard routes 1`] = `
Array [
  Array [
    Object {
      "url": "/fooga",
    },
    [Function],
  ],
  Array [
    Object {
      "url": "/fooga",
    },
    [Function],
  ],
]
`;

exports[`hrouter wildcards should register nested wildcard routes 1`] = `
Object {
  "fooga": Object {
    "*": Object {
      "fns": Array [
        [MockFunction],
      ],
    },
  },
}
`;

exports[`hrouter wildcards should register nested wildcard routes w/ multiple callbacks 1`] = `
Object {
  "*": Object {
    "fns": Array [
      [MockFunction],
      [MockFunction],
      [MockFunction],
    ],
  },
  "fooga": Object {
    "*": Object {
      "fns": Array [
        [MockFunction],
        [MockFunction],
      ],
    },
  },
}
`;

exports[`hrouter wildcards should register wildcard routes 1`] = `
Object {
  "*": Object {
    "fns": Array [
      [MockFunction],
    ],
  },
}
`;

exports[`hrouter wildcards should register wildcard routes w/ multiple callbacks 1`] = `
Object {
  "*": Object {
    "fns": Array [
      [MockFunction],
      [MockFunction],
      [MockFunction],
    ],
  },
}
`;
